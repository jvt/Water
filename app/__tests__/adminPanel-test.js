import {
	Text
} from 'react-native';
import React from 'react';
import AdminPanel from '../adminPanel';
import Button from 'apsl-react-native-button';
import Renderer from 'react-test-renderer';

describe('AdminPanel', () => {	
	test('BackButton', () => {
		const instance  = Renderer.create(
			<Button 
			style={{}}>
			Back
			</Button>

		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('TitleText', () => {
		const instance  = Renderer.create(
			<Text 
			style={{}}>
			Title:
			</Text>

		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('UsernameText', () => {
		const instance  = Renderer.create(
			<Text 
			style={{}}>
			Username:
			</Text>

		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('BannedTrueText', () => {
		const instance  = Renderer.create(
			<Text 
			style={{}}>
			Banned: True
			</Text>

		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('BannedFalseText', () => {
		const instance  = Renderer.create(
			<Text 
			style={{}}>
			Banned: False
			</Text>

		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('RoleUserText', () => {
		const instance  = Renderer.create(
			<Text 
			style={{}}>
			Role: User
			</Text>

		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('RoleWorkerText', () => {
		const instance  = Renderer.create(
			<Text 
			style={{}}>
			Role: Worker
			</Text>

		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('RoleManagerText', () => {
		const instance  = Renderer.create(
			<Text 
			style={{}}>
			Role: Manager
			</Text>

		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('RoleAdminText', () => {
		const instance  = Renderer.create(
			<Text 
			style={{}}>
			Role: Admin
			</Text>
		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

});
