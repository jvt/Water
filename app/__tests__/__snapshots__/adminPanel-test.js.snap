exports[`AdminPanel BackButton 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "center",
      "marginBottom": 10,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "fontSize": 18,
          "textAlign": "center",
        },
        undefined,
      ]
    }>
    Back
  </Text>
</View>
`;

exports[`AdminPanel BannedFalseText 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}>
  Banned: False
</Text>
`;

exports[`AdminPanel BannedTrueText 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}>
  Banned: True
</Text>
`;

exports[`AdminPanel RoleAdminText 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}>
  Role: Admin
</Text>
`;

exports[`AdminPanel RoleManagerText 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}>
  Role: Manager
</Text>
`;

exports[`AdminPanel RoleUserText 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}>
  Role: User
</Text>
`;

exports[`AdminPanel RoleWorkerText 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}>
  Role: Worker
</Text>
`;

exports[`AdminPanel TitleText 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}>
  Title:
</Text>
`;

exports[`AdminPanel UsernameText 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}>
  Username:
</Text>
`;
